### PR TITLE
Infrastructure to run JIT-compiled pipelines.

### PR DESCRIPTION
**Pull request by ryzhyk**
- Implemented serialization/deserialization APIs for JIT'd input and output handles.  These rely on JIT-compiled serializers and deserializers instead of serde.

- Refactored the server to move circuit initialization to the circuit thread (the thread that calls `step` and `kill`), since JIT-compiled circuits are not `Send`.

- Added a binary target to the `adapters` crate to run pipelines in JIT mode.

- A framework for integration tests that run JIT-compiled pipelines using the above-mentioned binary.

TODOs:
- Support CSV (de)serialization - pending on `dataflow-jit` to implement that.

- More integration tests - requires CSV support + a few other JIT features under construction

### CODE REVIEW

**Comment by ryzhyk** [comment]
Addresses #678

File: crates/adapters/src/server/mod.rs [hunk 1]
```diff
@@ -799,6 +816,8 @@ async fn output_endpoint(
                         // validated by `add_output_endpoint`.
                         .unwrap()
                         .neighborhood_descr_handle
+                        .as_ref()
+                        .ok_or_else(|| PipelineError::NeighborhoodNotSupported)?
```
Comment (Kixiron) [inline crates/adapters/src/server/mod.rs:820]:
```suggestion
                        .ok_or(PipelineError::NeighborhoodNotSupported)?
```

File: crates/adapters/src/server/mod.rs [hunk 2]
```diff
@@ -822,6 +841,8 @@ async fn output_endpoint(
                         .output_handles(&config.stream)
                         .unwrap()
                         .num_quantiles_handle
+                        .as_ref()
+                        .ok_or_else(|| PipelineError::QuantilesNotSupported)?
```
Comment (Kixiron) [inline crates/adapters/src/server/mod.rs:845]:
```suggestion
                        .ok_or(PipelineError::QuantilesNotSupported)?
```

**Review by Kixiron — APPROVED** [review]

File: crates/adapters/src/jit/json.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+/// Build JSON serializer configuration for specified layout and table schema.
+pub(crate) fn build_json_ser_config(layout: LayoutId, table_schema: &TableSchema) -> JsonSerConfig {
+    let mappings = table_schema
+        .fields
+        .iter()
+        .enumerate()
+        .map(|(index, column)| (index, column.name.to_string()))
+        .collect();
+    JsonSerConfig { layout, mappings }
+}
```
[the remark this answers is not in the document]
Comment (ryzhyk) [reply crates/adapters/src/jit/json.rs:30]: yeah, I'll fix it up after #748 has been merged.

**Pull request merged**